python_modules/dagster/dagster/_annotations.py

```diff
@@ -208,6 +208,7 @@ def deprecated_param(
     breaking_version: str,
     additional_warn_text: Optional[str] = ...,
     emit_runtime_warning: bool = ...,
+    require_param_in_signature: bool = True,
```

Comment:
nit: maybe something in the direction of `param_handled_via_kwargs`? With the current name I feel like you have to take a couple mental hops to "it doesnt need to be in the signature" to the end goal of "hide this arg via kwargs"